src/jsep.js

```diff
@@ -637,171 +802,115 @@ let jsep = function(expr) {
- * @param {string} op_name The name of the binary op to add
- * @param {number} precedence The precedence of the binary op (can be a float)
- * @return jsep
- */
-jsep.addBinaryOp = function(op_name, precedence) {
-	max_binop_len = Math.max(op_name.length, max_binop_len);
-	binary_ops[op_name] = precedence;
-	return this;
-};
-
-/**
- * @method jsep.addIdentifierChar
- * @param {string} char The additional character to treat as a valid part of an identifier
- * @return jsep
- */
-jsep.addIdentifierChar = function(char) {
-	additional_identifier_chars.add(char);
-	return this;
-};
-
-/**
- * @method jsep.addLiteral
- * @param {string} literal_name The name of the literal to add
- * @param {*} literal_value The value of the literal
- * @return jsep
- */
-jsep.addLiteral = function(literal_name, literal_value) {
-	literals[literal_name] = literal_value;
-	return this;
-};
-
-/**
- * @method jsep.removeUnaryOp
- * @param {string} op_name The name of the unary op to remove
- * @return jsep
- */
-jsep.removeUnaryOp = function(op_name) {
-	delete unary_ops[op_name];
-	if (op_name.length === max_unop_len) {
-		max_unop_len = getMaxKeyLen(unary_ops);
-	}
-	return this;
-};
-
-/**
- * @method jsep.removeAllUnaryOps
- * @return jsep
- */
-jsep.removeAllUnaryOps = function() {
-	unary_ops = {};
-	max_unop_len = 0;
-
-	return this;
-};
-
-/**
- * @method jsep.removeIdentifierChar
- * @param {string} char The additional character to stop treating as a valid part of an identifier
- * @return jsep
- */
-jsep.removeIdentifierChar = function(char) {
-	additional_identifier_chars.delete(char);
-	return this;
-};
-
-
-/**
- * @method jsep.removeBinaryOp
- * @param {string} op_name The name of the binary op to remove
- * @return jsep
- */
-jsep.removeBinaryOp = function(op_name) {
-	delete binary_ops[op_name];
-
-	if (op_name.length === max_binop_len) {
-		max_binop_len = getMaxKeyLen(binary_ops);
-	}
-
-	return this;
-};
-
-/**
- * @method jsep.removeAllBinaryOps
- * @return jsep
- */
-jsep.removeAllBinaryOps = function() {
-	binary_ops = {};
-	max_binop_len = 0;
-
-	return this;
-};
-
-/**
- * @method jsep.removeLiteral
- * @param {string} literal_name The name of the literal to remove
- * @return jsep
- */
-jsep.removeLiteral = function(literal_name) {
-	delete literals[literal_name];
-	return this;
-};
-
-/**
- * @method jsep.removeAllLiterals
- * @return jsep
- */
-jsep.removeAllLiterals = function() {
-	literals = {};
-
-	return this;
-};
-
+}
+
+// Backward Compatibility (before adding the static fields):
+const jsep = Jsep.parse;
+const staticMethods = Object.getOwnPropertyNames(Jsep);
+staticMethods
+	.slice(staticMethods.indexOf('version'), staticMethods.indexOf('removeAllLiterals') + 1)
+	.forEach((m) => {
+		jsep[m] = Jsep[m];
+	});
 export default jsep;
+
+
+// Static fields:
+Object.assign(Jsep, {
+	// Node Types
+	// ----------
+	// This is the full set of types that any JSEP node can be.
+	// Store them here to save space when minified
```

Comment:
I just noticed this comment. If the only purpose of these constants is to "save space when minified", then we should get rid of them, per #131. What do you think?

I'm only referring to the expression type constants. The character code ones are obviously useful.
Not sure e.g. `MEMBER_EXP` is easier to remember than `"MemberExpression"`. The only benefit I see is that perhaps we can change the type without having to change it everywhere in the code, but are we ever going to do that? 🤔

Reply:
I'm indifferent... there are some of the defines that are used multiple times, and others only once. For consistency it's kind of nice to have them here. And I think it's nicer to use a constant/enum rather than needing the exact string in several places.